frontend/src/services/transactions/request-manager.ts

```diff
@@ -139,14 +139,15 @@ export async function getRequestIdentifier(
     provider,
   );
   const receipt = await provider.waitForTransaction(transactionHash, 1);
-  if (receipt) {
+
+  if (receipt && receipt.status === 1) {
     const event = contract.interface.parseLog(receipt.logs[0]);
     if (event) {
       return event.args.requestId;
     }
   }
 
-  throw new Error("Request Failed. Couldn't retrieve Request ID");
+  throw new Error('Transaction reverted on chain.');
```

Comment:
i think the case where a receipt is undefined and a receipt status is 0 are two different cases right?

1. When its undefined it means that the request id was not found for this transaction.
2. When `receipt.status === 0` it means that the transaction was reverted.

Reply:
How do you end up with an undefined for a transactionReceipt?

Reply:
Check this out: https://docs.ethers.org/v5/api/providers/provider/#Provider-waitForTransaction

I don't know how to interpret this part: `If confirms is 0, this method is non-blocking and if the transaction has not been mined returns null.`

Does it return null when transaction has not been mined or does it only return null when confirms is 0 AND transaction has not been mined?